External/FEXCore/Source/Interface/Core/ObjectCache/ObjectCacheService.h

```diff
@@ -128,17 +235,102 @@ namespace FEXCore::CodeSerialize {
 
         class WorkItemRemoveNamedRegion : public NamedRegionWorkItem {
           public:
-            WorkItemRemoveNamedRegion(uint64_t base, uint64_t size, CodeRegionMapType::iterator entry)
+            WorkItemRemoveNamedRegion(uint64_t base, uint64_t size, std::unique_ptr<CodeRegionEntry> entry)
               : NamedRegionWorkItem {NamedRegionJobType::JOB_REMOVE_NAMED_REGION}
               , Base {base}
               , Size {size}
-              , Entry {entry} {}
+              , Entry {std::move(entry)} {}
 
             uint64_t Base;
             uint64_t Size;
-            CodeRegionMapType::iterator Entry;
+            std::unique_ptr<CodeRegionEntry> Entry;
         };
       /**  @} */
+
+    private:
+      NamedRegionObjectHandler *NamedRegionHandler;
+      CodeObjectSerializeService *CodeObjectCacheService;
+  };
+
+  class NamedRegionObjectHandler final {
+    public:
+      NamedRegionObjectHandler(FEXCore::Context::Context *ctx);
+
+      void HandleNamedRegionObjectJobs();
+
+      CodeObjectSerializationConfig const &GetDefaultSerializationConfig() const {
+        return DefaultSerializationConfig;
+      }
+
+    protected:
+      friend class AsyncJobHandler;
+
+      // Return a default code header based off the default serialization config
+      CodeObjectSerializationHeader DefaultCodeHeader(uint64_t Base, uint64_t Offset) const {
+        CodeObjectSerializationHeader Header{};
```

Comment:
This seems like a good place to use designated initializers. That way, it's harder to forget updating this code when adding new members in future changes, since the compiler can warn about it (e.g. gcc does this with `-Wextra`).
```suggestion
CodeObjectSerializationHeader Header{
        .Config = DefaultSerializationConfig,
        .OriginalBase = Base;
        .OriginalOffset = Offset;
        .NumCodeEntries = 0;
        .NumRelocationsTo = 0;
        .TotalRelocationsCount = 0;
};
```

Reply:
Resolved to use designated initializers.